base/src/main/java/org/aya/core/term/MetaLitTerm.java

```diff
@@ -0,0 +1,29 @@
+// Copyright (c) 2020-2022 Tesla (Yinsen) Zhang.
+// Use of this source code is governed by the MIT license that can be found in the LICENSE.md file.
+package org.aya.core.term;
+
+import kala.collection.immutable.ImmutableSeq;
+import kala.control.Option;
+import kala.tuple.Tuple2;
+import org.aya.core.def.GenericDef;
+import org.aya.core.repr.AyaShape;
+import org.aya.core.repr.ShapeRecognition;
+import org.aya.util.error.SourcePos;
+import org.jetbrains.annotations.NotNull;
+
+public record MetaLitTerm(
+  @NotNull SourcePos sourcePos,
+  @NotNull Object repr,
+  @NotNull ImmutableSeq<Tuple2<GenericDef, ShapeRecognition>> candidates,
```

Comment:
ImmutableMap?

Reply:
I think it is possible that the same data matches different shapes.
Maybe `ImmutableMap<GenericDef, ImmutableSeq<ShapeRecognition>>` ?

Reply:
I doubt. This represents the "candidates" which I do not expect to have so many.

But also since it's just a list of candidates, I tend to agree with the current code